### PR TITLE
add support for pagination (limit, offset, count) over API /rules and /vrules

### PR DESCRIPTION
**Pull request by AlvaroVega**
issue: https://github.com/telefonicaid/perseo-fe/issues/364

- [x] read limit, offset, count
- [x] use it in mongo methods
- [x] documentation
- [x] Tested

### CODE REVIEW

File: lib/routes/rulesController.js [hunk 1]
```diff
@@ -29,7 +29,10 @@ var rules = require('../models/rules'),
     myutils = require('../myutils');
 
 function GetAllRules(req, resp) {
-    rules.FindAll(req.service, req.subservice, function(err, data) {
+    var limit = req.query.limit ? req.query.limit : 1000;
```
Comment (fgalan) [inline lib/routes/rulesController.js]: Are they new parameters for the plain rules API? In that case, they should documented.

Comment (AlvaroVega) [reply lib/routes/rulesController.js]: New section about add documentation was added to this issue

Comment (AlvaroVega) [reply lib/routes/rulesController.js]: fixed in 13c5b6d687194887fbc85ae977d6589c90ef170f

File: lib/models/rulesStore.js [hunk 2]
```diff
@@ -111,6 +111,8 @@ function findAll(service, subservice, callback) {
             function(col, cb) {
                 col.find(criterion).toArray(function(err, rules) {
```
Comment (AlvaroVega) [inline lib/models/rulesStore.js]: better replace with: 
```
col.find(criterion).sort().limit(limit).skip(offset).toArray(function(err, rules) {
```
?

**Comment by AlvaroVega** [comment]
this is ready to merge

File: documentation/api.md [hunk 3]
```diff
@@ -76,3 +76,15 @@ CRUD for Visual Rules. Originally detailed in [DCA documentation](https://colabo
 | ------ | :----------------------- | ----------------------------------- |
```
Comment (fgalan) [inline documentation/api.md]: Related issue #364 already cited in CNR?

Comment (AlvaroVega) [reply documentation/api.md]: fixed in https://github.com/telefonicaid/perseo-fe/pull/396/commits/fe4fda790b23f1c76c75ebd65cbcaff5ecb9236d

File: documentation/api.md [hunk 4]
```diff
@@ -76,3 +76,15 @@ CRUD for Visual Rules. Originally detailed in [DCA documentation](https://colabo
...
+### Pagination
+
+Perseo-fe implements a pagination mechanism in order to help clients to retrieve large sets of resources. This mechanism
+works for all listing operations in the API (GET /rules, GET /vrules)
+
+The mechanism is based on three URI parameters:
+
+-   **limit**: in order to specify the maximum number of elements (default is 20, maximum allowed is 1000).
+-   **offset**: in order to skip a given number of elements at the beginning (default is 0)
+-   **count** (as option): if activated then a Fiware-Total-Count header is added to the response, with a count of total
```
Comment (fgalan) [inline documentation/api.md]: Is count really used?

Looking into the PR I only see req.query.limit and req.query.offset but not req.query.options, although maybe it was implemented in a past PR.

Comment (AlvaroVega) [reply documentation/api.md]: I'm going to open a new PR to include options=count since already count is always wired to true.

Comment (AlvaroVega) [reply documentation/api.md]: fixed in https://github.com/telefonicaid/perseo-fe/pull/396/commits/14b89e5c4f35337872f48f2f8b7000071e968eaf

File: lib/models/rulesStore.js (hunk 2 above)
Comment (fgalan) [reply lib/models/rulesStore.js]: Checked at skype. NTC.

**Pull request merged**